element_array_ephys/ephys_chronic.py

```diff
@@ -174,115 +189,149 @@ class EphysFile(dj.Part):
 
-                probe_electrodes = {key['electrode']: key
-                                    for key in electrode_query.fetch('KEY')}
+                probe_electrodes = {
+                    key["electrode"]: key for key in electrode_query.fetch("KEY")
+                }
 
                 electrode_group_members = [
                     probe_electrodes[channel_idx]
-                    for channel_idx in probe_data.ap_meta['channels_indices']]
+                    for channel_idx in probe_data.ap_meta["channels_indices"]
+                ]
             else:
                 raise NotImplementedError(
-                    'Processing for neuropixels'
-                    ' probe model {} not yet implemented'.format(probe_data.probe_model))
-
-            self.insert1({
-                **key,
-                **generate_electrode_config(probe_type, electrode_group_members),
-                'acq_software': acq_software,
-                'sampling_rate': probe_data.ap_meta['sample_rate'],
-                'recording_datetime': probe_data.recording_info['recording_datetimes'][0],
-                'recording_duration': np.sum(probe_data.recording_info['recording_durations'])})
-
-            root_dir = find_root_directory(get_ephys_root_data_dir(),
-                                probe_data.recording_info['recording_files'][0])
-            self.EphysFile.insert([{**key,
-                                    'file_path': fp.relative_to(root_dir).as_posix()}
-                                   for fp in probe_data.recording_info['recording_files']])
+                    "Processing for neuropixels"
+                    " probe model {} not yet implemented".format(probe_data.probe_model)
+                )
+
+            self.insert1(
+                {
+                    **key,
+                    **generate_electrode_config(probe_type, electrode_group_members),
+                    "acq_software": acq_software,
+                    "sampling_rate": probe_data.ap_meta["sample_rate"],
+                    "recording_datetime": probe_data.recording_info[
+                        "recording_datetimes"
+                    ][0],
+                    "recording_duration": np.sum(
+                        probe_data.recording_info["recording_durations"]
+                    ),
+                }
+            )
+
+            root_dir = find_root_directory(
+                get_ephys_root_data_dir(),
+                probe_data.recording_info["recording_files"][0],
+            )
+            self.EphysFile.insert(
+                [
+                    {**key, "file_path": fp.relative_to(root_dir).as_posix()}
+                    for fp in probe_data.recording_info["recording_files"]
+                ]
+            )
             # explicitly garbage collect "dataset"
             # as these may have large memory footprint and may not be cleared fast enough
```

Comment:
Is this empirically confirmed to improve memory performance?

Reply:
@dimitri-yatsenko , yes, I've tested and confirmed that the explicit garbage collection at this step does improve memory performance.
Without this step, calling the `make()` on this table for 6 consecutive times crashes a 32GB RAM server. Adding this garbage collection here completely eliminated that problem.

Reply:
Thanks @ttngu207!  This highlights weaknesses of the Python interpreter.